diff --git a/src/Encoder/CsvEncoder.php b/src/Encoder/CsvEncoder.php
index a3291f39cdee485a780c466098b0a6e9900044aa..5926b1603046f5d71e4b9224635eea87cdc39547 100644
--- a/src/Encoder/CsvEncoder.php
+++ b/src/Encoder/CsvEncoder.php
@@ -103,14 +103,14 @@ class CsvEncoder implements EncoderInterface, DecoderInterface {
   /**
    * {@inheritdoc}
    */
-  public function supportsEncoding($format) {
+  public function supportsEncoding(string $format): bool {
     return $format == static::$format;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function supportsDecoding($format) {
+  public function supportsDecoding(string $format): bool {
     return $format == static::$format;
   }
 
@@ -119,7 +119,7 @@ class CsvEncoder implements EncoderInterface, DecoderInterface {
    *
    * Uses HTML-safe strings, with several characters escaped.
    */
-  public function encode($data, $format, array $context = []) {
+  public function encode(mixed $data, string $format, array $context = []): string {
     switch (gettype($data)) {
       case "array":
         break;
@@ -288,7 +288,7 @@ class CsvEncoder implements EncoderInterface, DecoderInterface {
    * @throws \League\Csv\Exception
    * @throws \League\Csv\Exception
    */
-  public function decode($data, $format, array $context = []) {
+  public function decode(string $data, string $format, array $context = []) {
     $csv = Reader::createFromString($data);
     $csv->setDelimiter($this->delimiter);
     $csv->setEnclosure($this->enclosure);
